### PR TITLE
new: HitErrorMeter

### PR DESCRIPTION
**Pull request by KapiWilq**
## HitErrorMeter by KapiWilq

<a href="https://github.com/KapiWilq" target="_blank"></a>  

|||
| ------------- | ------------- |
| For | ingame, obs-overlay |
| Compatible with | tosu |
| Size |  720x100 |

> Supports all rulesets! | Very customizable! | The overlay resolution/size is for the default settings (both the overlay and the in-game score meter) in the osu!mania ruleset with EZDT at OD0.

hiterrormeter by kapiwilq osu_ruleset

### CODE REVIEW

File: counters/HitErrorMeter by KapiWilq/index.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
+import WebSocketManager from './js/socket.js';
+import HitErrorMeter from './js/hitErrorMeter.js';
+const socket = new WebSocketManager(`${location.host}`);
```
Comment (cyperdark) [inline counters/HitErrorMeter by KapiWilq/index.js:3]: const socket = new WebSocketManager(location.host);

File: counters/HitErrorMeter by KapiWilq/styles/main.css [hunk 2]
```diff
@@ -0,0 +1,172 @@
+@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@500&display=swap");
```
Comment (cyperdark) [inline counters/HitErrorMeter by KapiWilq/styles/main.css:1]: store font locally

File: counters/HitErrorMeter by KapiWilq/js/hitErrorMeter.js [hunk 3]
```diff
@@ -0,0 +1,644 @@
...
+
+        let segmentForTheTick = hitError <= 0 ? this.getHitWindowSegment(-hitError, 'Early') : this.getHitWindowSegment(hitError, 'Late');
+        let tickPositionPercentage = this.getTickPositionPercentage(Math.abs(hitError));
+
+        let tick = document.createElement('div');
+        tick.classList.add('tick');
+
+        // We don't know if hit error segments are hidden - extract RGB(A) values and set the opacity manually.
+        // RegExp source: https://regex101.com/library/dVOwn0
+        const RGBA_REGEXP = /rgba?\((?<r>[.\d]+)[, ]+(?<g>[.\d]+)[, ]+(?<b>[.\d]+)(?:\s?[,\/]\s?(?<a>[.\d]+%?))?\)/;
```
Comment (cyperdark) [inline counters/HitErrorMeter by KapiWilq/js/hitErrorMeter.js:578]: you can avoid regex by using this:
```js
const [r, g, b, a] = text.replace('rgba(', '').replace('rgb(', '').replace('(', '').split(',').map(r => r.trim());
```


optional, need test

File: counters/HitErrorMeter by KapiWilq/js/hitErrorMeter.js [hunk 4]
```diff
@@ -0,0 +1,644 @@
...
+            //     https://github.com/ppy/osu/pull/25415
+            if (this.client === 'lazer' || (this.client === 'stable' && this.mods.includes('v2')))
+                this.hitWindows = {
+                    hit320: (this.overallDiff <= 5 ? 22.4 - 0.6 * this.overallDiff : 24.9 - 1.1 * this.overallDiff) * this.rate,
+                    hit300: (64 - 3 * this.overallDiff) * this.rate,
+                    hit200: (97 - 3 * this.overallDiff) * this.rate,
+                    hit100: (127 - 3 * this.overallDiff) * this.rate,
+                    hit50: (151 - 3 * this.overallDiff) * this.rate
+                };
+            else if (this.client === 'mania' && this.isConvert)
```
Comment (cyperdark) [inline counters/HitErrorMeter by KapiWilq/js/hitErrorMeter.js:292]: client === mania?

**Review by cyperdark — CHANGES_REQUESTED** [review]

**Comment by KapiWilq** [comment]
drafting this until I get mania to be actually accurate ([see discussion](https://discord.com/channels/1056534107330445362/1284013170368843787/1311004610558955611))

**Comment by KapiWilq** [comment]
should be ready to go now

**Pull request merged**